.circleci/config.yml

```diff
@@ -0,0 +1,172 @@
+        - ~/.m2
+
+  save_gems_cache: &save_gems_cache
+    save_cache:
+      key: *gems_key
+      paths:
+        - vendor/bundle
+
+  ## Dependencies
+  ruby_yum: &ruby_yum
+    run:
+      name: Install Ruby Dev Libs
+      command: yum install -y ruby-devel
+
+  bundler_update: &bundler_update
+    run:
+      name: Update bundler
+      command: gem install bundler --version 1.16.3
+
+  ruby_dependencies: &ruby_dependencies
+    run:
+      name: Download Ruby Dependencies
+      command: bundle check || bundle install --path vendor/bundle
+
+jobs:
+  test_unit:
+    <<: *android_config
+    steps:
+      - checkout
+      - *restore_gradle_cache
+      - *restore_gems_cache
+      - *ruby_yum
+      - *bundler_update
+      - *ruby_dependencies
+      - *save_gradle_cache
+      - *save_gems_cache
+      - run:
+          name: Run unit tests
+          command: bundle exec fastlane unit_tests
+      - store_artifacts:
+          path: app/build/reports/
+          destination: /reports/
+      - store_test_results:
+          path: app/build/test-results/
+          destination: /test-results/
+
+  lint_check:
+    <<: *android_config
+    steps:
+      - checkout
+      - *restore_gradle_cache
+      - *restore_gems_cache
+      - *ruby_yum
+      - *bundler_update
+      - *ruby_dependencies
+      - *save_gradle_cache
+      - *save_gems_cache
+      - run:
+          name: Run Lint checks
+          command: bundle exec fastlane lint_check
+
+  send_coverage_report:
+    <<: *android_config
+    steps:
+      - checkout
+      - *restore_gradle_cache
+      - *restore_gems_cache
+      - *ruby_yum
+      - *bundler_update
+      - *ruby_dependencies
+      - *save_gradle_cache
+      - *save_gems_cache
+      - run:
+          name: Send reports to coveralls.io
+          command: ./gradlew cobertura coveralls
+
+  deploy_firebase:
+    <<: *android_config
+    steps:
+      - checkout
+      - *restore_gradle_cache
+      - *restore_gems_cache
+      - *ruby_yum
+      - *bundler_update
+      - *ruby_dependencies
+      - *save_gradle_cache
+      - *save_gems_cache
+      - run:
+          name: Deploy to Firebase App Distribution
+          command: bundle exec fastlane deploy_to_firebase
+
+  publish_bintray:
+    <<: *android_config
+    steps:
+      - checkout
+      - *restore_gradle_cache
+      - *restore_gems_cache
+      - *ruby_yum
+      - *bundler_update
+      - *ruby_dependencies
+      - *save_gradle_cache
+      - *save_gems_cache
+      - run:
+          name: Publish to Bintray
+          command: ./gradlew bintrayUpload
+
+workflows:
+  version: 2
+  build_test_deploy:
+    jobs:
+      - test_unit
+
+      - send_coverage_report:
+          requires:
+            - test_unit
+
+      - lint_check:
+          requires:
+            - test_unit
+
+      - deploy_firebase:
+          filters:
+            branches:
+              only:
+                - master
+          requires:
+            - test_unit
```

Comment:
We should remove the requirement since the test unit job will be removed

Reply:
I did all the changes mentioned above